### PR TITLE
Geometry bounds are updated with the projection.

### PR DESCRIPTION
**Pull request by dabhicusp**
In the current implementation, we are utilizing [`geometry.bounds().getInfo()`](https://github.com/google/Xee/blob/main/xee/ext.py#L282-L284) to retrieve the bound points. However, this approach yields inconsistent results when the `latitude value` of a point exceeds `104`, resulting in the return of a `constant latitude value`. So I am adding the projection into the bound points to fix this issue.

Example:
```
ee_image = ee.Image("projects/anthromet-prod/assets/1km_uk_nimrod_composite_rainfall_radar/202202120220_nimrod_ng_radar_rainrate_composite_1km_UK")

scale = 5000
projection = ee.Projection("EPSG:4326", [1, 0, 0, 0, -1, 0]).atScale(scale)
ee_image = ee_image.reproject(projection)

point = ee.Geometry.Point(-110.2414893624401 ,104)
distance = 311.5
geometry = point.buffer(distance, proj=projection).bounds(proj=projection)

ic = ee.ImageCollection([ee_image])
ds = xarray.open_dataset(
    ic, projection=ee_image.projection(), geometry=geometry,
    ee_mask_value=-99999, engine = 'ee'
)
ds
```
The above ds returns only a `single latitude point` when utilizing the current code. However, upon updating the bounds with `bounds(proj = projection)`, the ds accurately returns 623 latitude points.

Thank you @bbabenko for suggesting the above apporoach.

### CODE REVIEW

**Review by naschmitz — COMMENTED** [review]
Could you add some integration tests that make this fail before the changes?

**Comment by dabhicusp** [comment]
@naschmitz Test cases for the above have been added.

File: xee/ext.py [hunk 1]
```diff
@@ -279,9 +279,14 @@ def get_info(self) -> Dict[str, Any]:
       rpcs.append(('projection', self.projection))
 
     if isinstance(self.geometry, ee.Geometry):
-      rpcs.append(('bounds', self.geometry.bounds()))
+      rpcs.append(('bounds', self.geometry.bounds(1, proj=self.projection)))
```
Comment (naschmitz) [inline xee/ext.py:282]: If the user specifies an `ee.Geometry`, we should just honor it. No need to transform an existing geometry.

Comment (naschmitz) [reply xee/ext.py:282]: My mistake. I thought we were passing the geometry in directly without calling bounds on it.

**Comment by lbferreira** [comment]
I think that if the user don't provide the projection the code will break. I'm not sure, but I think I noted it when I was implementing a very similar change in a previous PR: https://github.com/google/Xee/pull/127

**Comment by dabhicusp** [comment]
@lbferreira If user not passed any projection code still works but returns the wrong output which current  code returns.
So in that particular situation, would it be better to use the projection of the image or is the current implementation okay to use?
What you say @naschmitz @jdbcode.

Code:
```
ee_image = ee.Image("projects/anthromet-prod/assets/1km_uk_nimrod_composite_rainfall_radar/202202120220_nimrod_ng_radar_rainrate_composite_1km_UK")
scale = 5000
projection = ee.Projection("EPSG:4326", [1, 0, 0, 0, -1, 0]).atScale(scale)
ee_image = ee_image.reproject(projection)
point = ee.Geometry.Point(-110.2414893624401 ,104)
distance = 311.5
geometry = point.buffer(distance, proj=projection).bounds(proj=projection)

# No projection
print(geometry.bounds(1, proj=None).getInfo())

# Projection from the image
print(geometry.bounds(1, proj=ee_image.projection()).getInfo())

# Projection passed explicitely.
print(geometry.bounds(1, proj=self.projection).getInfo())
```

**Comment by jdbcode** [comment]
> geometry = ee.Geometry.Rectangle([[0, -80], [360, 80]],None,geodesic=False)

Geom constructors use EPSG:4326 as default CRS, which has longitude range of -180 to 180 with the prime meridian at 0. You are specifying  geometry whose east limit is the prime meridan and wraps around the globe fully, crossing the antimeridian before meeting itself back at the prime meridian. Crossing the antimeridian is likely to give unexpected results. If you are trying to extract data in such a way that the center is at -180 (the antimeridian) you may need to download the data into two parts (east of the antimeridian and west of the antimeridian) and put them together client-side (see [antimeridian cutting](https://datatracker.ietf.org/doc/html/rfc7946#section-3.1.9) in the GeoJSON docs), or use a CRS whose center is the antimeridian (maybe something like [EPSG:3832](https://epsg.io/3832)?).

**Review by jdbcode — APPROVED** [review]

**Comment by jdbcode** [comment]
I still worry that it will introduce coordinate precision issues that might affect some people. But if it fixes your case, that's great, and if it causes issues for other people then we'll have confirmed cases to help us figure out a better solution.

**Pull request merged**